src/parser.rs

```diff
@@ -765,11 +765,11 @@ impl<'a> Scanner<'a> {
             if lref.contains(&b'$') {
                 self.unget = ltok;
                 break;
-            } else if !is_valid_label(lref) {
+            } else if is_valid_label(lref) {
+                self.labels.push(ltok);
```

Comment:
I find it very weird to have the success case in the _middle_ of an else-chain.  Usually I prefer the end, sometimes I'll do the beginning, but I actively avoid the middle